pyVim/context.py

```diff
@@ -0,0 +1,181 @@
+        container, viewType, recursive)
+
+    children = containerView.view
+    for child in children:
+        print("VM Name: ", child.summary.config.name)
+
+    """
+
+    def __init__(self, host='localhost', port=443, user='root', pwd='',
+                 service="hostd", adapter="SOAP", namespace=None, path="/sdk",
+                 version=None, keyFile=None, certFile=None, thumbprint=None,
+                 sslContext=None, b64token=None, mechanism='userpass'):
+        """
+        Most values are optional. Specify the minimum values for your use case.
+        :param host: default 'localhost' specify your host here
+        :param port: default '443' specify only if different
+        :param user: default 'root' specify only if needed
+        :param pwd: default '' specify only if different
+        :param service: defaults to 'hostd' specify only if you need to
+        :param adapter: defaults to SOAP specify only if required to do so
+        :param namespace: default will be calculated on connection
+        :param path: defaults to /sdk only specify if you must
+        :param version: default calculates the highest version number SDK + Host can use together
+        :param keyFile: specify only if using self-signed PEM key files
+        :param certFile: specify only if using self-signed certs
+        :param thumbprint: specify only if overriding based on known SSL thumbprints
+        :param sslContext: specify only if you are overriding default SSL context behaviors
+        :param b64token: specify if you have a Token such as the HoK Token
+        :param mechanism: specify as either 'userpass' or 'sspi'
+        """
+        self.si = None
+        self.open_count = 0
+        self.host = host
+        self.port = port
+        self.user = user
+        self.pwd = pwd
+        self.service = service
+        self.adapter = adapter
+        self.namespace = namespace
+        self.path = path
+        self.version = version
+        self.keyFile = keyFile
+        self.certFile = certFile
+        self.thumbprint = thumbprint
+        self.sslContext = sslContext
+        self.b64token = b64token
+        self.mechanism = mechanism
+
+    def __enter__(self):
+        self.open_count += 1
+        if self.si is not None:
+            return self.si
+        self.si = open(host=self.host, port=self.port, user=self.user, pwd=self.pwd, service=self.service,
+                       adapter=self.adapter, namespace=self.namespace, path=self.path, version=self.version,
+                       keyFile=self.keyFile, certFile=self.certFile, thumbprint=self.thumbprint,
+                       sslContext=self.sslContext, b64token=self.b64token, mechanism=self.mechanism)
+        return self.si
+
+    def __exit__(self, *args):
+        if (self.open_count == 1):
+            close(self.si)
+            self.si = None
+        self.open_count -= 1
+        return
+
+
+def open(host='localhost', port=443, user='root', pwd='',
```

Comment:
It seems open and close functions are similar as Connect/Disconnect in connect.py beside SetSi(). I wonder if it's better to add these two methods in connect.py so we could reuse the existing logic?